src/api/uploads/FolderUploadNode.js

```diff
@@ -66,7 +71,12 @@ class FolderUploadNode {
         this.parentFolderId = parentFolderId;
 
         await this.createAndUploadFolder(errorCallback, isRoot);
+        if (!this.getFolderId()) {
```

Comment:
FYI - this does not use this.folderId because I need to be able to mock it for tests.